### ISSUE TITLE
Quiet music

### ISSUE THREAD

**Issue by wisetrap11** [dhewm/dhewm3#502]
Hi,
I've just finished playing through Doom 3 using this source port. However, something that's been bugging me throughout my playthrough was that the music was noticeably quieter in the port compared to its volume when playing Doom 3 without the port, to the point where I'm not even sure if there was any music at all outside of the main theme and the music that plays during the Cyberdemon intro and defeat cutscenes. I should note that everything else (sound effects, voice lines, etc.) is perfectly audible-- It's just the music that's having this problem. I've tried changing the Speakers and EAX 4.0 HD settings in the options, but to no avail, and the settings show that the volume slider is all the way up. As such, I'm not sure how to fix this issue. Is there any way to solve this?

**Comment by DanielGibson** [comment]
Are you sure there even is other music in the game?
If so, do you have a video or sth of the original where the music is audible, for comparison?

**Comment by adecrepitcabbage** [comment]
@DanielGibson I think it's important to listen to end-users when they come to a place as relatively niche as a git to file a report. For us, all that matters is trying to preserve the experience as close to 'as-is' as possible. Right now, it's still possible to play the original Doom 3 and note things these things; for some of us, myself included, the auditory experience is one of the prime reasons Doom 3 is our favourite entry into the series.

For the record, there is music in the GUI, various audio 'stingers' that contribute **_greatly_** to the ambience, and change the feel dramatically when removed or dampened. I don't know if I'd call them all music, but ambience works just as well. The most notable early example that isn't the pause menu music is the first confrontation with the Pinky, and the most notable non-music variations are the effects and chatter that occur during the scene at the Comms tower where you first see a Lost Soul at work and a bunch of different sfx (channels?) are playing at once. I do not possess the technology or internet speed to capture videos, because my internet is worse then it was 20 years ago.

I will be lending my full report to #503 which I believe this is a sub-issue of. My hope is that in doing so I will be able to contribute to a future permanent non-user-dependent fix to the issue as reported.

**Comment by DanielGibson** [comment]
> I think it's important to listen to end-users when they come to a place as relatively niche as a git to file a report.

Sure, but keep in mind that I'm doing this for free, in my (scarce) free time.
So I won't spend lots of time comparing different Doom3 versions on different systems myself, trying to find a difference *(that might even only occur on some places in some levels!)*, but instead ask for clarifications and demonstrations of the issues.

Videos that demonstrate the differences, like the ones posted in #503, are very helpful for this, and so are savegames and/or step-by-step instructions on how to reproduce bugs.

In this specific case, the only place where there's *real* music is the main menu, in the game there are "only" ambient noises that, unlike "normal/real" music in many other games (think Classic Doom or Quake 2) are *not* played in stereo directly to the player (like if the Doom guy was wearing headphones) - those ambient noises come from sound sources *that have a position* in the levels, so naturally they get louder/quiter (or even become inaudible) depending on your position etc.
So to reproduce and investigate this bug, I'd need to know a few specific places in specific levels where the differences are noticeable, and ideally a description on what's different, so I know what to look (or rather listen) for.

**Comment by ZT-1** [comment]
I have also noticed this. A big one is the EPD section in alpha labs 1. There's an ambient track there that is barely audible with dhewm3 compared to the base game. I've noticed that it is very slightly more audible with EAX off, but with an extremely slight difference. I assume this is also the case with most if not all of the ambient tracks. It is especially noticeable with this one however in alpha labs 1.

dhewm3
https://github.com/user-attachments/assets/c71ad578-eece-4da6-8568-0c292e478dc4
Base game
https://github.com/user-attachments/assets/79068563-57d0-421c-9564-bf93d6aff2b7

**Comment by DanielGibson** [comment]
Thanks, that should help reproducing the issue!

For the base game, did you use the Windows Version?
If so, did it use OpenAL or the old/original sound backend?

**Comment by ZT-1** [comment]
Yes, this is on Windows 11. The unmodified game on Steam. 

I'm pretty sure it does use OpenAL, which should be the old/original backend? Whatever the default game on Steam is using.

**Comment by ZT-1** [comment]
The audio file of that particular section is hellsbells2.ogg in pak003>sound>ed>music. Not sure if that helps.

Also, I've noticed that dhewm3 is quieter in general than the default game. The volume slider at max in dhewm3 is equal to the volume slider at a significantly lower place in the default game.

**Comment by DanielGibson** [comment]
Not sure if Doom3 on Windows uses OpenAL by default, will *at least* depend on whether it can find an OpenAL32.dll on the system, but I'm not even sure if it was enabled by default.
The `s_useOpenAL` CVar is (only) set to `1` after starting Vanilla Doom3 if OpenAL is used.

> I've noticed that dhewm3 is quieter in general than the default game. 

Yeah that's a feature to prevent clipping when too many loud sounds play at once.
The Dhewm3Settings menu (can usually be opened with F10) has settings for this.

**Comment by DanielGibson** [comment]
Oh right, a CVar that you could try on dhewm3 is `s_quadraticFalloff 0`

**Comment by ZT-1** [comment]
> Oh right, a CVar that you could try on dhewm3 is `s_quadraticFalloff 0`

Didn't seem to do much regarding ambient music. Just made the game slightly louder.

**Comment by ZT-1** [comment]
If I disable "Scale down and clamp all sound volumes" (`s_scaleDownAndClamp`), then I can hear that music track pretty clearly. A lot better. All of the other sounds become a lot louder like the default game of course if you do that though.

Should I just do this then? It also says in the info box that I should disable `s_alOutputLimiter`. What should that be set to? It is at -1 by default.

**Comment by DanielGibson** [comment]
Well `s_scaleDownAndClamp` should make *everything* in the game quieter and sound pretty much the same as without it (just quieter) - except when there are lots of loud noises that would either cause clipping or would make make OpenAL temporarily reduce the overall volume.

So the problem apparently isn't that the music is more quiet than the rest of the game and not audible because of that, but that the whole game is quiet?
So you could just turn up the master volume on the operating systems sound mixer or the speakers?

> Should I just do this then? It also says in the info box that I should disable s_alOutputLimiter. What should that be set to? It is at -1 by default.

`0` disables it, `1` enables it, `-1` doesn't set anything so the default from OpenAL (or the setting in its config) is used

**Comment by ZT-1** [comment]
Yes, scaledownandclamp seems to make everything a lot quieter, thus likely making the music inaudible too. Maybe it thinks that would cause clipping. 

Well not really about cranking up the volume. You could have the volume of everything at max with scaledownandclamp enabled and the music will still be way too quiet.
